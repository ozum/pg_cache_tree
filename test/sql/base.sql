\set ECHO none
BEGIN;
\i sql/pg_cache_tree.sql

CREATE SCHEMA other;

CREATE TABLE other."Account" (
  id INTEGER,
  "parentId" INTEGER,
  "parentsCache" INTEGER [] DEFAULT '{}'::integer[] NOT NULL,
  "childrenCache" INTEGER [] DEFAULT '{}'::integer[] NOT NULL,
  CONSTRAINT account_pkey PRIMARY KEY(id)
);

SELECT ct_create_trigger('other."Account"');

INSERT INTO other."Account" (id, "parentId") VALUES
(1, null),
(2, null),
(3, null),
(10, 1),
(100, 10),
(101, 10),
(20, 2),
(21, 2),
(200, 20),
(201, 20),
(210, 21)
;

UPDATE other."Account" SET "parentId" = 21 WHERE id = 200;
UPDATE other."Account" SET id = 25, "parentId" = 21 WHERE id = 201;

DELETE FROM other."Account" WHERE id = 210;

\set ECHO all

SELECT * FROM other."Account" ORDER BY id;

ROLLBACK;
