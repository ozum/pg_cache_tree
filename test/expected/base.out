\set ECHO none
 ct_create_trigger 
-------------------
 
(1 row)

SELECT * FROM other."Account" ORDER BY id;
 id  | parentId | parentsCache | childrenCache  
-----+----------+--------------+----------------
   1 |          | {}           | {10,101,100}
   2 |          | {}           | {21,25,200,20}
   3 |          | {}           | {}
  10 |        1 | {1}          | {101,100}
  20 |        2 | {2}          | {}
  21 |        2 | {2}          | {25,200}
  25 |       21 | {2,21}       | {}
 100 |       10 | {1,10}       | {}
 101 |       10 | {1,10}       | {}
 200 |       21 | {2,21}       | {}
(10 rows)

ROLLBACK;
